compiler/res/prog8lib/atari/textio.p8

```diff
@@ -118,7 +118,28 @@ asmsub  scroll_down  (ubyte alsocolors @ Pc) clobbers(A)  {
 }
 
 
-romsub $FFD2 = chrout(ubyte char @ A)    ; TODO
+romsub $F2B0 = outchar(ubyte char @ A)
+romsub $F2Fd = waitkey()
+
+asmsub chrout(ubyte char @ A) {
```

Comment:
you should add "clobbers(A, X, Y)" to the declaration.  Then you can omit the saving/restoring of all the registers.

Reply:
agree! even tried it, but somehow it didn't work as expected. will give it another try later.

Reply:
If I understand it correctly, clobbers works when the routine is called in a user program. So, when I use chrout in another %asm routine this won't help me and I still have to save the registers myself.

The better solution is to re-write the print routine using Atari Central I/O (CIO). This is the clean and compatible way to do things. But maybe it's better to first extend syslib.p8 for the atari. I would at least need all the OS calls and CIO labels for this to work.

Reply:
Youre 100% correct